cmd/kube-apiserver/app/options/validation.go

```diff
@@ -50,8 +52,8 @@ func validateClusterIPFlags(options *ServerRunOptions) []error {
 	// Complete() expected to have set Primary* and Secondary*
 	// primary CIDR validation
 	var ones, bits = options.PrimaryServiceClusterIPRange.Mask.Size()
-	if bits-ones > 20 {
-		errs = append(errs, errors.New("specified --service-cluster-ip-range is too large"))
+	if bits-ones > maxCIDRbits {
+		errs = append(errs, fmt.Errorf("specified --service-cluster-ip-range is too large. Network CIDR should not be bigger than /%d", bits-maxCIDRbits))
```

Comment:
do I make a CIDR range smaller by making the number following the slash bigger? is it confusing to say "should not be bigger than /%d" when the solution is to increase that number?

Reply:
yeah, these things are always confusing, I'm quoting this definition that things that express it correctly:

> A short prefix allows for more addresses -- and, therefore, forms a bigger block -- while a longer prefix indicates less addresses and a smaller block.

But what will be the right english wording?

Reply:
I would find it easier to understand something that mapped directly to the numbers I had to provide, like 

```go
fmt.Errorf("specified --service-cluster-ip-range is too large; for %d-bit addresses, the mask must be >= /%d", bits, bits-maxCIDRbits)
```

it might also help to factor this aspect into a helper method:
```go
func validateMaxCIDRRange(cidr, maxRangeBits) error { ... }
```
to exercise in tests and use this way:
```go
if err :=  validateMaxCIDRRange(options.PrimaryServiceClusterIPRange, maxCIDRbits); err != nil {
  errs = append(errs, fmt.Errorf("specified --service-cluster-ip-range is too large: %w", err))
}
```

then we could have unit tests showing the actual messages printed to help make sure they are correct and read naturally